README.md

```
@@ -137,45 +223,175 @@ $kind /= option-b
+### Diagnostic dump
+
+`CBOR.diag()` generates a hex-annotated view of the encoded bytes, with each field's
+type, offset, and decoded value shown inline. The comment column auto-adjusts to the
+widest line in the output, and long strings wrap across multiple lines rather than
+being truncated.
+
+```python
+print(CBOR({0: "test", 1: 42}).diag())
+```
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)